File: Bachelor/Semester6-Finals/MPP - Projecting and Programing Methods/tReservation-Lab1CS/tReservation-Lab1CS/repository/ISeatsRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using tReservation_Lab1CS.repository;

namespace tReservation_Lab1CS
{
    interface ISeatsRepository
    {
        void delete(SeatsHelper seatsHelper);
        List<Seats> findAllByCourse(int courseId);
        Seats findOneBySeatNumber(SeatsHelper seatsHelper);
        void save(Seats seats);
        void update(SeatsHelper seatsHelper, string seatClient);
    }
}

File: Bachelor/Semester6-Finals/MPP - Projecting and Programing Methods/tReservation-Lab1/src/main/java/repository/ISeatsRepository.java
package repository;

public interface ISeatsRepository<SeatsHelper,Seats> {
    int size();
    void save(Seats entity);
    void delete(SeatsHelper seatsHelper);
    void update(SeatsHelper seatsHelper,String clientName);
    Seats findOne(SeatsHelper seatsHelper);
    Iterable<Seats> findAll();
    Iterable<Seats> findAllByCourse(int courseId);
}
